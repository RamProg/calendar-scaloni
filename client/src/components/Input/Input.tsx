interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
  hasError: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input: React.FC<InputProps> = ({
  type,
  value,
  onChange,
  name,
  label,
  hasError,
}) => {
  return (
    <>
      <label className="block" id={name}>
        {label}
        <input
          name={name}
          type={type}
          value={value}
          onChange={onChange}
          className="block w-full mt-1"
        />
        {hasError && (
          <span className="text-red-500">
            There is an issue with the {label}
          </span>
        )}
      </label>
    </>
  );
};

export default Input;
