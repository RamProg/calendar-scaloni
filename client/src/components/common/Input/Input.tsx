interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label: string;
  hasError: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input: React.FC<InputProps> = ({
  type,
  value,
  onChange,
  name,
  label,
  hasError,
}) => {
  return (
    <>
      <label className="block mb-4 text-2xl font-bold sm:text-xl" id={name}>
        {label}
        <input
          name={name}
          type={type}
          value={value}
          onChange={onChange}
          className="block w-full p-2 mt-1 font-normal h-14 sm:h-10"
        />
        {hasError && (
          <span className="font-normal text-red-500" data-testid="error">
            There is an issue with the {label}
          </span>
        )}
      </label>
    </>
  );
};

export default Input;
