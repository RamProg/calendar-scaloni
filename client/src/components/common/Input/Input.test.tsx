import '@testing-library/jest-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import Input from './Input';

describe('Input', () => {
  const mockOnChange = jest.fn();
  const inputLabel = 'Test Input';

  test('renders correctly', () => {
    render(
      <Input
        label={inputLabel}
        name="test"
        type="text"
        value=""
        onChange={mockOnChange}
        hasError={false}
      />
    );
    const inputElement = screen.getByLabelText(inputLabel);
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toMatchSnapshot();
  });

  test('calls onChange prop when text is entered', () => {
    render(
      <Input
        label={inputLabel}
        name="test"
        type="text"
        value=""
        onChange={mockOnChange}
        hasError={false}
      />
    );

    const inputElement = screen.getByLabelText(inputLabel);

    fireEvent.change(inputElement, { target: { value: 'test value' } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  test('shows error state when hasError prop is true', () => {
    render(
      <Input
        label={inputLabel}
        name="test"
        type="text"
        value=""
        onChange={mockOnChange}
        hasError={true}
      />
    );
    const errorElement = screen.getByText(
      `There is an issue with the ${inputLabel}`
    );
    expect(errorElement).toBeInTheDocument();
  });
});
