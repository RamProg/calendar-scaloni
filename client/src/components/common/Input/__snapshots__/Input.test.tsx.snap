// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  class="block w-full p-2 mt-1 font-normal h-14 sm:h-10"
  name="test"
  type="text"
  value=""
/>
`;
